liquibase-core/src/main/java/liquibase/command/CommandDefinition.java

```diff
@@ -36,13 +36,8 @@ protected CommandDefinition(String[] name) {
         pipeline = new TreeSet<>((o1, o2) -> {
             final int order = Integer.compare(o1.getOrder(this), o2.getOrder(this));
             if (order == 0) {
-                // if o1 requires o2 dependencies it must come after o2
-                for (Class<?> required : o1.requiredDependencies()) {
-                    if (o2.providedDependencies().stream().anyMatch(required::equals)) {
-                        return 1;
-                    }
-                }
-                return o1.getClass().getName().compareTo(o2.getClass().getName());
+                // keep the provided order
```

Comment:
Why is the complex logic for dependency ordering no longer necessary?

Reply:
It's because the order has been already calculated before adding the items to the pipeline. This new sort was in fact messing up the calculateddependency order by sorting it alphabetically.